File: test_project/gen_code_cs/bag.cs
namespace demo
{

    
    public class Bag
    {
        
        
        public readonly int Id;
        
        
        public readonly string Name;
        
        
        public readonly System.Collections.Generic.List<int> PriceList;
        
        
        public readonly System.Collections.Generic.Dictionary<string, int> PropMap;
        
        
        public Bag(int id, string name, System.Collections.Generic.List<int> pricelist, System.Collections.Generic.Dictionary<string, int> propmap)
        {
            
            this.Id = id;
            
            this.Name = name;
            
            this.PriceList = pricelist;
            
            this.PropMap = propmap;
            
        }
    }

}

File: test_project/gen_code_ts/bag.ts
export namespace demo {


    export class Bag {


    public readonly Id: number;


    public readonly Name: string;


    public readonly PriceList: number[];


    public readonly PropMap: Map<string, number>;


    constructor(Id: number, Name: string, PriceList: number[], PropMap: Map<string, number>) {


        this.Id = Id;

        this.Name = Name;

        this.PriceList = PriceList;

        this.PropMap = PropMap;

    }
}

}
